libs/cosmos-sdk/store/mpt/store_iterator_wrap.go

```diff
@@ -0,0 +1,65 @@
+package mpt
+
+import (
+	"bytes"
+	"sort"
+
+	ethstate "github.com/ethereum/go-ethereum/core/state"
+)
+
+// wrapIterator is a wrap of mpt iterator which can be iterated by the origin key.
+// It is compatible with cachekv.
+type wrapIterator struct {
+	*mptIterator
+
+	cacheKeys [][]byte
+}
+
+func newWrapIterator(t ethstate.Trie, start, end []byte) *wrapIterator {
+	var keys [][]byte
+	mptIter := newOriginIterator(t, start, end)
+	for ; mptIter.Valid(); mptIter.Next() {
+		keys = append(keys, mptIter.Key())
```

Comment:
the  may be can filter the key not  in the range of the start and end
'for ; mptIter.Valid(); mptIter.Next() {

}
'